spock-core/src/main/java/org/spockframework/compiler/WhereBlockRewriter.java

```diff
@@ -116,9 +121,6 @@
       .forEach(expression -> stats.previous());
 
     if (potentialHeaderRow.size() > 1) {
-      if (!potentialHeaderRow.stream().allMatch(VariableExpression.class::isInstance)) {
```

Comment:
why remove the fail fast check?

Reply:
Because it is not too nice to have it in the result.
I actually added this when adding the semicolon-separator for data tables and it also only affected them as it was not added for pipe-separated data tables.
Now that in the current implementation I morph the pipe-separated table to a semicolon-separated table to be able to have unified handling and cross-multiplication, it affect both.
And the result is quite ugly.
Like it was before I removed it, you get for the following:
```groovy
where:
a | 1 | b
1 | 1 | 1
```
six compile errors.
First it is morphed to
```groovy
where:
a ; 1 ; b
1 ; 1 ; 1
```
Then it checks `a ; 1 ; b`, sees the `1`, adds a compile error, discards one statement (which is just the `a`) and continues with the next statement.
That means now it finds `1 ; b`, sees the `1`, adds a compile error, discards one statement and continues with the next statement.
Now it finds `b`, adds a compile error as just `b` is not a valid parameterization for the where block, discards one statement and continues with the next statement.
Now it finds `1 ; 1 ; 1` and again considers this a header row as no table was successfully started parsing yet, sees the `1`s, adds a compile error, discards one statement and continues.
Then the same game with `1 ; 1`.
And as last compile error the "not a parameterization" due to just seeing `1`.

This could be improved of course.
If not only one statement is discarded, but the whole row for example.
Then it would see `a ; 1 ; b`, complain about the `1`, discard the row and continue with the next statement.
That means it sees `1 ; 1 ; 1` as potential header row, sees the `1`s and adds another complaint that in the header row only variable names are allowed.

What could maybe be done is to discard the whole table if a non-variable in the header row is found.
But that would then need according parsing and so on and if anything goes wrong there it might hide the actual error accidentally and so on. So I decided to throw out this pre-check and just let it be handled in the transformation to simple data pipes where it was checked anyway too.